gadget/layout.go

```diff
@@ -214,26 +208,28 @@ func layoutVolumeStructures(volume *Volume) (structures []LaidOutStructure, byNa
 		structures[idx] = ps
 	}
 
+	// Check:
+	// - No structure overlaps
+	// - offset-write meets the restrictions defined in the gadget
+	// TODO Reuse fully validateCrossVolumeStructure
 	previousEnd := quantity.Offset(0)
 	for idx, ps := range structures {
 		if ps.StartOffset < previousEnd {
-			return nil, nil, fmt.Errorf("cannot lay out volume, structure %v overlaps with preceding structure %v", ps, structures[idx-1])
+			return nil, fmt.Errorf("cannot lay out volume, structure %v overlaps with preceding structure %v", ps, structures[idx-1])
 		}
 		previousEnd = ps.StartOffset + quantity.Offset(ps.VolumeStructure.Size)
 
-		offsetWrite, err := resolveOffsetWrite(ps.VolumeStructure.OffsetWrite, byName)
-		if err != nil {
-			return nil, nil, fmt.Errorf("cannot resolve offset-write of structure %v: %v", ps, err)
+		if err := validateOffsetWrite(ps.VolumeStructure, structures[0].VolumeStructure, volume.MinSize()); err != nil {
+			return nil, err
 		}
-		structures[idx].AbsoluteOffsetWrite = offsetWrite
 	}
 
-	return structures, byName, nil
+	return structures, nil
```

Comment:
If `byName` is being defined and allocated inside this function, then only assigned to in one place and not used elsewhere or returned, is it still necessary?

Reply:
Good catch, indeed it was not needed anymore